### PR TITLE
AstroNav GPS Cartridge

### PR DESCRIPTION
**Pull request by ArchRBX**
## About the PR
Adds a new PDA cartridge that gives your PDA the ability to act as a handheld GPS.

Salvage Specialists start with this cartridge preinstalled, and a spare cartridge is available in the Quartermaster's locker.

## Why / Balance
Plenty of salvage specialists forget to pick up a GPS and die cold and alone out in space. This cartridge provisions salvagers with at least the basics of making it back to the station alive.

## Technical details

Cartridge uses the HandheldGPS component so any GPS changes will also work with this. (e.g. #31814)

Refactors the HandheldGPS component to be stored in Shared.

## Media

https://github.com/user-attachments/assets/17ef40b9-60b1-4cd0-84f8-173e9427f68a

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

  - SharedHandheldGPSComponent merged into to HandheldGPSComponent, moved to Shared
  - namespace Content.Client.GPS.Components changed to Content.Shared.GPS.Components

**Changelog**

:cl:
- add: The AstroNav PDA cartridge has now been added, which turns your PDA into a handheld GPS
- add: The AstroNav cartridge has been added to the QM locker, and comes preinstalled on salvage PDA's

### CODE REVIEW

**Comment by lzk228** [comment]
how does it look in pda ui?

**Review by slarticodefast — DISMISSED** [review]

File: Content.Server/CartridgeLoader/Cartridges/AstroNavCartridgeSystem.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        SubscribeLocalEvent<AstroNavCartridgeComponent, CartridgeRemovedEvent>(OnCartridgeRemoved);
+    }
+
+    private void OnCartridgeAdded(Entity<AstroNavCartridgeComponent> ent, ref CartridgeAddedEvent args)
+    {
+        var loader = args.Loader;
+        if (!EntityManager.HasComponent<HandheldGPSComponent>(loader))
+        {
+            EntityManager.AddComponent<HandheldGPSComponent>(loader);
+        }
```
[the remark this answers is not in the document]
Comment (ArchRBX) [reply Content.Server/CartridgeLoader/Cartridges/AstroNavCartridgeSystem.cs:25]: Good to know, thank you!

**Comment by ArchRBX** [comment]
> how does it look in pda ui?

image

If there's a better-fitting sprite I'd be happy to incorporate it.

**Comment by Plykiya** [comment]
was kinda hoping this would be a PDA app and not just an item status thing

**Comment by deltanedas** [comment]
i was doing that then gave up

**Comment by Plykiya** [comment]
> i was doing that then gave up

the cycle of PDA app development...

**Comment by JIPDawg** [comment]
> i was doing that then gave up

Don't give up Delta, I believe in you. You're our only hope. Our savior. Please!

**Comment by deltanedas** [comment]
the gargantuan effort to have text be in a larger font size

**Comment by ArchRBX** [comment]
> was kinda hoping this would be a PDA app and not just an item status thing

It's still a PDA app, just one that doesn't open out into its own UI. This PR leaves room for that in future PR's if people wanna try. Didn't wanna include it in the scope of this PR as it would affect balance a lot more depending on what was provided to them. Having a mass scanner PDA app on hand at all times would be *awesome* but also change the balance a fair bit.

File: Resources/Prototypes/Entities/Objects/Devices/pda.yml [hunk 2]
```diff
@@ -400,6 +400,14 @@
     accentVColor: "#8900c9"
   - type: Icon
     state: pda-miner
+  - type: CartridgeLoader
+    uiKey: enum.PdaUiKey.Key
+    notificationsEnabled: false
```
Comment (juliangiebel) [inline Resources/Prototypes/Entities/Objects/Devices/pda.yml:405]: Why are you disabling notifications by default here?

**Review by juliangiebel — DISMISSED** [review]
The item status doesn't update when inserting or ejecting the cartridge without installing it.
It should at least get updated when the cartridge gets removed without having been installed.

File: Resources/Prototypes/Entities/Objects/Devices/pda.yml (hunk 2 above)
Comment (ArchRBX) [reply Resources/Prototypes/Entities/Objects/Devices/pda.yml:405]: Great question! Left over from debugging. Easy fix.

**Comment by ArchRBX** [comment]
> The item status doesn't update when inserting or ejecting the cartridge without installing it. It should at least get updated when the cartridge gets removed without having been installed.

The latter is fairly easy to fix, but the first is an issue with the statuscontrol system that I'm not sure how to fix currently. There's no way to force-trigger a hand status control update at the moment. Only adding stuff to hands actually triggers the update.

**Pull request merged**